client/app/components/dashboards/TextboxDialog.jsx

```diff
@@ -40,11 +40,31 @@ function TextboxDialog({ dialog, isNew, ...props }) {
     });
   }, [dialog, isNew, text]);
 
+  const confirmDialogDismiss = useCallback(() => {
+    const originalText = props.text;
+    if (text !== originalText) {
+      Modal.confirm({
+        title: "Confirm action",
+        content: "Closing this dialog will discard all changes you've made. Are you sure?",
+        okText: "Yes, discard changes",
+        okType: "danger",
+        cancelText: "No, get me back",
```

Comment:
```suggestion
        cancelText: "Cancel",
```
(Is this the default?)